modules/samples/connectedcup/component/api/src/main/java/org/coffeeking/api/ConnectedCupControllerServiceImpl.java

```diff
@@ -20,73 +20,98 @@
-        return Response.ok().entity("Coffee ordered.").build();
     }
 
-    public ConnectedCupMQTTConnector getConnectedCupMQTTConnector() {
-        return ConnectedCupControllerServiceImpl.connectedCupMQTTConnector;
-    }
+    @Path("stats/{deviceId}/sensors/{sensorName}")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Response getDeviceStats(@PathParam("deviceId") String deviceId, @PathParam("sensorName") String sensor,
+                                               @QueryParam("from") long from, @QueryParam("to") long to) {
+        String fromDate = String.valueOf(from);
+        String toDate = String.valueOf(to);
+        String query = " deviceId:" + deviceId + " AND deviceType:" +
+                ConnectedCupConstants.DEVICE_TYPE + " AND time : [" + fromDate + " TO " + toDate + "]";
+        String sensorTableName = getSensorEventTableName(sensor);
 
-    public void setConnectedCupMQTTConnector(
-            final ConnectedCupMQTTConnector connectedCupMQTTConnector) {
-
-        Runnable connector = new Runnable() {
-            public void run() {
-                if (waitForServerStartup()) {
-                    return;
-                }
-                ConnectedCupControllerServiceImpl.connectedCupMQTTConnector = connectedCupMQTTConnector;
-                if (MqttConfig.getInstance().isEnabled()) {
-                    connectedCupMQTTConnector.connect();
-                } else {
-                    log.warn("MQTT disabled in 'devicemgt-config.xml'. " +
-                             "Hence, ConnectedCupMQTTConnector not started.");
-                }
+        try {
+            if (!APIUtil.getDeviceAccessAuthorizationService().isUserAuthorized(new DeviceIdentifier(deviceId,
+                       ConnectedCupConstants.DEVICE_TYPE))) {
+                return Response.status(Response.Status.UNAUTHORIZED.getStatusCode()).build();
             }
-        };
-        Thread connectorThread = new Thread(connector);
-        connectorThread.setDaemon(true);
-        connectorThread.start();
+            List<SensorRecord> sensorDatas;
+            List<SortByField> sortByFields = new ArrayList<>();
+            SortByField sortByField = new SortByField("time", SORT.ASC, false);
+            sortByFields.add(sortByField);
+            sensorDatas = APIUtil.getAllEventsForDevice(sensorTableName, query, sortByFields);
+            return Response.ok().entity(sensorDatas).build();
+        } catch (AnalyticsException e) {
+            String errorMsg = "Error on retrieving stats on table " + sensorTableName + " with query " + query;
+            log.error(errorMsg);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).entity(errorMsg).build();
+        } catch (DeviceAccessAuthorizationException e) {
+            log.error(e.getErrorMessage());
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
```

Comment:
This exception is thrown when there is an error occurred when calling the authorization service,eg: plugin or group management initialization failure. So this would raise due to an internal issue. So isn't the response Internal Server Error is the correct one ?